tests/test_mcp_compliance.py

```
@@ -0,0 +1,308 @@
+        endpoints = mcp_config.get_endpoints()
+        mcp_url = endpoints.get("mcp_server")
+
+        if not mcp_url or "localhost" not in mcp_url:
+            pytest.skip("MCP server not running locally")
+
+        try:
+            async with aiohttp.ClientSession() as session:
+                # Test health endpoint
+                async with session.get(
+                    f"{mcp_url}/health", timeout=aiohttp.ClientTimeout(total=5)
+                ) as response:
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)